website/content/docs/sync/github.mdx

```diff
@@ -149,6 +150,99 @@ unintended disclosure of secrets in any GitHub Action outputs. The following sni
 
 If the GitHub destination uses the default `secret-key` granularity, the values are masked by GitHub automatically.
 
+## GitHub application
+
+Instead of authenticating with a personal access token, you can choose to
+authenticate with a
+[custom GitHub application](https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app).
+
+Start by following the GitHub instructions for
+[installing a GitHub app](https://docs.github.com/en/apps/using-github-apps/installing-your-own-github-app).
+to install your GitHub application on a specified repository and note the
+assigned installation ID.
+
+<Tip title="Your installation ID is in the app URL">
+
+  You can find your assigned installation ID in the URL path parameter:
+  `https://github.com/settings/installations/<INSTALLATION_ID>`
+
+</Tip>
+
+Then add your GitHub application to your Vault instance.
+
+To use your GitHub application with Vault:
+
+- the application must have permission to read and write secrets to Vault.
+- you must know the application ID assigned by GitHub.
+- you must generate a private key on GitHub.
+
+Callback and redirect URLs are not required at this time.
+
+To configure the application in Vault, use `vault write` with the
+`sys/sync/github-apps` endpoint to assign a unique name and set the relevant
+information:
+
+```shell-session
+$ vault write sys/sync/github-apps/<APP_NAME> \
+  app_id=<APP_ID> \
+  private_key=<PATH_TO_PRIVATE_KEY>
+```
+
+For example:
+
+<CodeBlockConfig hideClipboard>
+
+
+```shell-session
+$ vault write sys/sync/github-apps/my-app-1 \
+  app_id=12345678 \
+  private_key=@${PATH_TO_PRIVATE_KEY}
+
+Key            Value
+---            -----
+app_id         12345678
+fingerprint    <fingerprint>
```

Comment:
Is there an easy way to verify the fingerprint we can suggest?

Reply:
Addressed most of your feedback except this one! Can you clarify? Do we mean via the Github UI, or a cli command? Or both?

Reply:
Any or both. We return this fingerprint so the logical question for a user is "what do I do with this, what is the process for me to verify this fingerprint so I'm sure my private key was not tempered with on its way to Vault?"?